### PR TITLE
Merge and rename packages and components to HttpSys*

### PR DESCRIPTION
**Pull request by Tratcher**
#283 

Stage 1: merge packages, namespaces, tests, initial renames, reduced object visibility

Future work:
- internal refactoring to reduce object model duplication across layers
- perf optimizations possible via deeper object model integration

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.HttpSys/RequestProcessing/ResponseBody.cs [hunk 1]
```diff
@@ -10,11 +10,11 @@
...
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
-using static Microsoft.Net.Http.Server.UnsafeNclNativeMethods;
+using static Microsoft.AspNetCore.Server.HttpSys.UnsafeNclNativeMethods;
 
-namespace Microsoft.Net.Http.Server
+namespace Microsoft.AspNetCore.Server.HttpSys
 {
-    internal class ResponseStream : Stream
+    internal class ResponseBody : Stream
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.HttpSys/RequestProcessing/ResponseBody.cs]: Just renaming for fun?

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.HttpSys/RequestProcessing/ResponseBody.cs]: There were two ResponseSteam classes. I may be able to de-duplicate later.

File: src/Microsoft.AspNetCore.Server.HttpSys/WebHostBuilderWebListenerExtensions.cs [hunk 2]
```diff
@@ -3,47 +3,44 @@
...
         /// The Microsoft.AspNetCore.Hosting.IWebHostBuilder to configure.
         /// </param>
         /// <returns>
         /// The Microsoft.AspNetCore.Hosting.IWebHostBuilder.
         /// </returns>
-        public static IWebHostBuilder UseWebListener(this IWebHostBuilder hostBuilder)
+        public static IWebHostBuilder UseHttpSys(this IWebHostBuilder hostBuilder)
         {
             return hostBuilder.ConfigureServices(services => {
-                services.AddTransient<IConfigureOptions<WebListenerOptions>, WebListenerOptionsSetup>();
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.HttpSys/WebHostBuilderWebListenerExtensions.cs]: It was used to set up a logger, which can now be passed directly from DI.

File: test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/project.json [hunk 3]
```diff
@@ -23,6 +23,7 @@
     },
     "net451": {
       "frameworkAssemblies": {
+        "System.DirectoryServices": "",
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/project.json]: What APIs are you using that you weren't using before (or did I just miss that).

Comment (Tratcher) [inline test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/project.json]: This isn't new, it's just merged from the other test project.

File: src/Microsoft.AspNetCore.Server.HttpSys/Constants.cs [hunk 4]
```diff
@@ -13,6 +13,7 @@ internal static class Constants
         internal const string Close = "close";
         internal const string Zero = "0";
         internal const string SchemeDelimiter = "://";
+        internal const int Status101SwitchingProtocols = 101;
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Server.HttpSys/Constants.cs]: Should this be moved to StatusCodes in HttpAbstractions?

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.HttpSys/Constants.cs]: Kestrel could use it too: https://github.com/aspnet/KestrelHttpServer/search?utf8=%E2%9C%93&q=101

Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Server.HttpSys/Constants.cs]: https://github.com/aspnet/HttpAbstractions/pull/748, any other status codes we want to add?

**Comment by Tratcher** [comment]
@JunTaoLuo Your commit regressed the tests xplat https://github.com/aspnet/WebListener/commit/3b42433f07106225a8d1bde8e227990b82967a52. You forgot to make them conditional like all of the others.

**Comment by JunTaoLuo** [comment]
@Tratcher ah I see. Didn't realize those need to be updated.

**Pull request merged**